File: bank-service/src/main/java/rs/edu/raf/mapper/CardIssuerMapper.java
package rs.edu.raf.mapper;

import org.springframework.stereotype.Component;
import rs.edu.raf.dto.CardIssuerCreateDTO;
import rs.edu.raf.dto.CardIssuerDTO;
import rs.edu.raf.model.card.CardIssuer;

@Component
public class CardIssuerMapper {

    public CardIssuer cardIssuerCreateDTOtoCardIssuer(CardIssuerCreateDTO cardIssuerCreateDTO) {
        CardIssuer cardIssuer = new CardIssuer();

        cardIssuer.setName(cardIssuerCreateDTO.name());
        cardIssuer.setBIN(cardIssuerCreateDTO.BIN());
        cardIssuer.setMII(cardIssuerCreateDTO.MII());

        return cardIssuer;
    }

    public CardIssuerDTO cardIssuerToCardIssuerDTO(CardIssuer cardIssuer) {
        return new CardIssuerDTO(cardIssuer.getId(), cardIssuer.getName(), cardIssuer.getMII(), cardIssuer.getBIN());
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/mapper/CardIssuerMapper.kt
package org.example.bankservicekotlin.mapper

import org.example.bankservicekotlin.dto.CardIssuerCreateDTO
import org.example.bankservicekotlin.dto.CardIssuerDTO
import org.example.bankservicekotlin.model.card.CardIssuer
import org.springframework.stereotype.Component

@Component
class CardIssuerMapper {
    fun cardIssuerCreateDTOtoCardIssuer(cardIssuerCreateDTO: CardIssuerCreateDTO): CardIssuer {
        return CardIssuer(null, cardIssuerCreateDTO.name, cardIssuerCreateDTO.MII, cardIssuerCreateDTO.BIN, null)
    }

    fun cardIssuerToCardIssuerDTO(cardIssuer: CardIssuer): CardIssuerDTO {
        return CardIssuerDTO(cardIssuer.id, cardIssuer.name, cardIssuer.MII, cardIssuer.BIN)
    }
}